docs/docs/api/optimizers/GEPA/GEPA_Advanced.md

```diff
@@ -0,0 +1,435 @@
+
+````
+I provided an assistant with the following instructions to perform a task for me:
+```
+<curr_instructions>
+```
+
+The following are examples of different task inputs provided to the assistant along with the assistant's response for each of them, and some feedback on how the assistant's response could be better:
+```
+<inputs_outputs_feedback>
+```
+
+Your task is to write a new instruction for the assistant.
+
+Read the inputs carefully and identify the input format and infer detailed task description about the task I wish to solve with the assistant.
+
+Read all the assistant responses and the corresponding feedback. Identify all niche and domain specific factual information about the task and include it in the instruction, as a lot of it may not be available to the assistant in the future. The assistant may have utilized a generalizable strategy to solve the task, if so, include that in the instruction as well.
+
+Provide the new instructions within ``` blocks.
+````
+
+This template is automatically filled with:
+
+- `<curr_instructions>`: The current instruction being optimized
+- `<inputs_outputs_feedback>`: Structured markdown containing predictor inputs, generated outputs, and evaluation feedback
+
+Example of default behavior:
+
+```python
+# Default instruction proposer is used automatically
+gepa = dspy.GEPA(
+    metric=my_metric,
+    reflection_lm=dspy.LM(model="gpt-5", temperature=1.0, max_tokens=32000, api_key=api_key),
+    auto="medium"
+)
+optimized_program = gepa.compile(student, trainset=examples)
+```
+
+### When to Use Custom instruction_proposer
+
+**Note:** Custom instruction proposers are an advanced feature. Most users should start with the default proposer, which works well for most text-based optimization tasks.
+
+Consider implementing a custom instruction proposer when you need:
+
+- **Multi-modal handling**: Process images (dspy.Image) alongside textual information in your inputs
+- **Nuanced control on limits and length constraints**: Have more fine-grained control over instruction length, format, and structural requirements
+- **Domain-specific information**: Inject specialized knowledge, terminology, or context that the default proposer lacks and cannot be provided via feedback_func. This is an advanced feature, and most users should not need to use this.
+- **Provider-specific prompting guides**: Optimize instructions for specific LLM providers (OpenAI, Anthropic, etc.) with their unique formatting preferences
+- **Coupled component updates**: Handle situations where 2 or more components need to be updated together in a coordinated manner, rather than optimizing each component independently (refer to component_selector parameter, in Custom Component Selection section, for related functionality)
```

Comment:
Link to the component selection section?

Reply:
👍